test/Ansatz/Chain_test.jl

```diff
@@ -107,6 +107,36 @@
     @testset "Canonization" begin
         using Tenet
 
+        @testset "contract" begin
+            qtn = rand(Chain, Open, State; n = 5, p = 2, χ = 20)
+            canonized = canonize(qtn)
+
+            @test_throws ArgumentError contract(canonized, :between, Site(1), Site(2); direction = :dummy)
+            @test_throws ArgumentError contract!(canonized, :between, Site(1), Site(2); direction = :dummy)
+
+            canonized = canonize(qtn)
+
+            for i in 1:4
+                contract_some = contract(canonized, :between, Site(i), Site(i + 1))
+                Bᵢ = select(contract_some, :tensor, Site(i))
+
+                @test isapprox(contract(TensorNetwork(contract_some)), contract(TensorNetwork(qtn)))
+
+                @test length(tensors(canonized)) == length(tensors(contract_some)) + 1 # We removed one singular value vector
```

Comment:
This is not needed

Reply:
?

Reply:
When testing, you don't need to check every internal consequence, but the consequences on the interface mainly.